README.adoc

```diff
@@ -36,30 +38,238 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
+
+`src/main/java/hello/PubSubApplication.java`
+[source, java]
+----
+@SpringBootApplication
+public class PubSubApplication {
+
+  public static void main(String[] args) throws IOException {
+    SpringApplication.run(PubSubApplication.class, args);
+  }
+
+}
+----
+
+Additionally, since we're building a web application, you can also create a WebAppController class
+to separate between the controller and configuration logic, although they can also live in the same
+place.
+
+`src/main/java/hello/WebAppController.java`
+[source, java]
+----
+@RestController
+public class WebAppController {
+}
+----
+
+We're still missing two files: the HTML and the properties one.
+
+`src/main/resources/static/index.html`
+[source, java]
+----
+include::/complete/src/main/resources/static/index.html[]
+----
+
+`src/main/resources/application.properties`
+[source, java]
+----
+include::/complete/src/main/resources/application.properties[]
+----
 
-Create a new controller for your Spring application:
+The Spring Cloud GCP Core Boot starter can auto-configure these two properties and make them
+optional. Although if the properties are specified here, they overwrite the auto-configured values.
+The Spring Cloud GCP Core Boot starter is bundled with the Spring Cloud GCP Pub/Sub Boot starter.
 
-`src/main/java/hello/GreetingController.java`
+The GCP project ID is auto-configured from the GOOGLE_CLOUD_PROJECT environment variable, among
+https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L287[several other sources].
+The OAuth2 credentials are auto-configured from the
+https://developers.google.com/identity/protocols/application-default-credentials[GOOGLE_APPLICATION_CREDENTIALS]
+environment variable. If the https://cloud.google.com/sdk/[Google Cloud SDK] is installed, this
+environment variable is easily configured by running the `gcloud auth application-default login`
+command in the same process of the app, or a parent one.
+
+== Create an inbound channel adapter
+
+An inbound channel adapter listens to messages from a Google Cloud Pub/Sub subscription and sends
+them to a Spring channel in an application.
+
+Instantiating an inbound channel adapter requires a subscriber factory and the name of an existing
+subscription. The Spring Cloud GCP Pub/Sub Boot starter provides a pre-configured subscriber factory
+which you can simply inject as a method argument.
+
+`src/main/java/hello/PubSubApplication.java`
 [source,java]
 ----
-include::complete/src/main/java/hello/GreetingController.java[]
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=messageChannelAdapter]
 ----
 
-NOTE: The above example does not specify `GET` vs. `PUT`, `POST`, and so forth, because `@RequestMapping` maps all HTTP operations by default. Use `@RequestMapping(method=GET)` to narrow this mapping.
+After the channel adapter is instantiated, an output channel where the adapter sends the received
```

Comment:
This statement sounds too technical and just as a description of the code vs what the code is accomplishing like "Now, create an output channel to send the messages"